tests/optimize/test_backtesting.py

```diff
@@ -1239,3 +1239,88 @@ def test_backtest_start_multi_strat_nomock_detail(default_conf, mocker,
     assert 'BACKTESTING REPORT' in captured.out
     assert 'SELL REASON STATS' in captured.out
     assert 'LEFT OPEN TRADES REPORT' in captured.out
+
+
+@pytest.mark.filterwarnings("ignore:deprecated")
+def test_backtest_start_multi_strat_caching(default_conf, mocker, caplog, testdatadir):
+
+    default_conf.update({
+        "use_sell_signal": True,
+        "sell_profit_only": False,
+        "sell_profit_offset": 0.0,
+        "ignore_roi_if_buy_signal": False,
+    })
+    patch_exchange(mocker)
+    backtestmock = MagicMock(return_value={
+        'results': pd.DataFrame(columns=BT_DATA_COLUMNS),
+        'config': default_conf,
+        'locks': [],
+        'rejected_signals': 20,
+        'final_balance': 1000,
+    })
+    mocker.patch('freqtrade.plugins.pairlistmanager.PairListManager.whitelist',
+                 PropertyMock(return_value=['UNITTEST/BTC']))
+    mocker.patch('freqtrade.optimize.backtesting.Backtesting.backtest', backtestmock)
+    mocker.patch('freqtrade.optimize.backtesting', show_backtest_result=MagicMock())
+
+    path_glob = MagicMock(return_value=['not important'])
+    load_backtest_metadata = MagicMock(return_value={
+        'StrategyTestV2': {'run_id': '1'},
+        'TestStrategyLegacyV1': {'run_id': 'changed'}
+    })
+    load_backtest_stats = MagicMock(side_effect=[
+        {
+            'metadata': {'StrategyTestV2': {'run_id': '1'}},
+            'strategy': {'StrategyTestV2': {}},
+            'strategy_comparison': [{'key': 'StrategyTestV2'}]
+        },
+        {
+            'metadata': {'TestStrategyLegacyV1': {'run_id': '2'}},
+            'strategy': {'TestStrategyLegacyV1': {}},
+            'strategy_comparison': [{'key': 'TestStrategyLegacyV1'}]
+        }
+    ])
+    get_strategy_run_id = MagicMock(side_effect=['1', '2', '2'])
+    mocker.patch('pathlib.Path.glob', path_glob)
+    mocker.patch.multiple('freqtrade.data.btanalysis',
+                          load_backtest_metadata=load_backtest_metadata,
+                          load_backtest_stats=load_backtest_stats)
+    mocker.patch('freqtrade.misc.get_strategy_run_id', get_strategy_run_id)
```

Comment:
this mock is "missing" - which means it doesn't apply correctly.

When mocking a directly imported function (like get_strategy_run_id) - you'll have to replace it at the import location (freqtrade.optimize.backtesting - not where it's defined).

That mock is however also not applying correctly - most likely because the test-file is importing backtesting before the mock is actually running (which sometimes requires odd workarounds).

I remember i had this once before which led me to move the test to another file - but i can't remember which test that was ...